flower/utils/pg_storage.py

```diff
@@ -87,10 +114,22 @@ def close_connection():
 def get_all_events():
     logger.debug('Events loading from postgresql persistence backend')
     cursor = connection.cursor()
-    try:
-        cursor.execute(_all_events)
-        for row in cursor:
-            yield row[0]
-        logger.debug('Events loaded from postgresql persistence backend')
-    finally:
-        cursor.close()
+    retries_remaining = REQ_MAX_RETRIES
+    while True:
```

Comment:
The retry logic repeats here; it should be implemented in a more general way so that it can be reused in multiple places.

Reply:
the main difference is that write operations can rollback. I can probably factor that out into a flag.  priority 2 :)